src/common/mfu_util.c

```diff
@@ -1105,6 +1110,239 @@ bool mfu_is_lustre(const char* path)
+    char* found = strtok_r(line, " ", &saveptr);
+    while (found != NULL) {
+        num++;
+        switch (num) {
+            case 1:
+                mip->mfu_mi_source = MFU_STRDUP(found);
+                break;
+            case 2:
+                mip->mfu_mi_mountpoint = MFU_STRDUP(found);
+                break;
+            case 3:
+                mfu_mount_info_parse_fstype_line(found, strlen(found)+1, mip);
+                break;
+            case 4:
+                mip->mfu_mi_mountopts = MFU_STRDUP(found);
+                break;
+            default:
+                break;
+        }
+
+        found = strtok_r(NULL, " ", &saveptr);
+    }
+
+    return(0);
+}
+
+int mfu_mount_info_get_filesystem_list(struct list_head* mfu_mi_list)
+{
+    if (mfu_mi_list == NULL) {
+        return(EINVAL);
+    }
+
+    errno = 0;
+    FILE* mountfp = fopen("/proc/self/mounts", "r");
+    if (mountfp == NULL) {
+        return(errno);
+    }
+
+    char* line = NULL;
+    size_t len = 0;
+    ssize_t bytes = getline(&line, &len, mountfp);
+    while (bytes != -1)
+    {
+        struct mfu_mount_info* itemp = MFU_CALLOC(1, sizeof(struct mfu_mount_info));
+        if (itemp == NULL) {
+            fclose(mountfp);
+            return(ENOMEM);
+        }
+        INIT_LIST_HEAD(&itemp->mfu_mi_linkage);
+
+        /*
+         * Parse one of the lines in /proc/self/mounts
+         */
+        mfu_mount_info_parse_mount_line(line, len, itemp);
+
+        /*
+         *  Get the device for the mountpoint in question.
+         */
+        errno = 0;
+        struct stat st;
+        int rc = stat(itemp->mfu_mi_mountpoint, &st);
+        if (rc == -1) {
+            fclose(mountfp);
+            fprintf(stderr, "Could not stat: %s (%d)", itemp->mfu_mi_mountpoint, errno);
+            return(1);
+        }
+        itemp->mfu_mi_dev = st.st_dev;
+
+        /*
+         *  Now we can insert the item into the list.
+         */
+        list_add_tail(&itemp->mfu_mi_linkage, mfu_mi_list);
+        bytes = getline(&line, &len, mountfp);
+    }
+
+    fclose(mountfp);
+    return(0);
+}
+
+void mfu_init_filesystem_list(void)
+{
+    if (mfu_mi_list_initialized == true) {
+        MFU_LOG(MFU_LOG_INFO, "The filesystem list is already initialized...");
+        return;
+    }
+
+    MFU_LOG(MFU_LOG_INFO, "Initializing filesystem list...");
+    INIT_LIST_HEAD(&mfu_mi_list);
+
+    int rc = mfu_mount_info_get_filesystem_list(&mfu_mi_list);
+    if (rc != 0) {
+        struct mfu_mount_info* itemp = NULL;
+        struct mfu_mount_info* tmpitemp = NULL;
+        list_for_each_entry_safe(itemp,
+                                 tmpitemp,
+                                 &mfu_mi_list,
+                                 mfu_mi_linkage)
+        {
+            list_del_init(&itemp->mfu_mi_linkage);
+            mfu_mount_info_destroy(itemp);
+            mfu_free(&itemp);
+        }
+        return;
+    }
+
+    mfu_mi_list_initialized = true;
```

Comment:
I think mfu_mi_list_initialized should be set to true at line 1259. My original code accidentally had it set to true in both places accidentally. I know that the entire list has not been set up properly at that point, but if anything fails, then later on when we call mfu_destroy_filesystem_list(), nothing will actually get cleaned up.

Reply:
Ok, I moved this up.  I also set it back to false if the list gets deleted.  Would that still work?

Reply:
Yes. I that looks fine to me!